cdi-microservice-provider/src/main/java/io/silverware/microservices/providers/cdi/internal/MicroservicesCDIExtension.java

```diff
@@ -99,61 +90,50 @@ public void beforeBeanDiscovery(@Observes final BeforeBeanDiscovery beforeEvent,
 
       // Create a registry of Microservices
-      if (bean.getBeanClass().isAnnotationPresent(Microservice.class)) {
-         // we do not want to register the injection points
-         if (bean.getQualifiers().stream().filter(qualifier -> qualifier.annotationType().isAssignableFrom(MicroserviceReference.class)).count() == 0) {
-            final Microservice annotation = bean.getBeanClass().getAnnotation(Microservice.class);
-            final String microserviceName = annotation.value().length() > 0 ? annotation.value() : bean.getBeanClass().getSimpleName();
+      if (isMicroserviceBean(bean)) {
+         final Microservice annotation = bean.getBeanClass().getAnnotation(Microservice.class);
+         final String microserviceName = !annotation.value().isEmpty() ? annotation.value() : bean.getBeanClass().getSimpleName();
 
-            this.context.registerMicroservice(getMicroserviceMetaData(microserviceName, bean));
-         }
+         this.context.registerMicroservice(getMicroserviceMetaData(microserviceName, bean));
       }
 
       // Create proxies for the corresponding injection points
-      final Set<InjectionPoint> injectionPoints = bean.getInjectionPoints();
-      for (final InjectionPoint injectionPoint : injectionPoints) {
-         final Set<Annotation> annotations = injectionPoint.getAnnotated().getAnnotations();
-         injectionPoint.getQualifiers().stream().filter(qualifier -> MicroserviceReference.class.isAssignableFrom(qualifier.annotationType())).forEach(qualifier -> {
-            final Member member = injectionPoint.getMember();
-            if (member instanceof Field) {
-               if (log.isTraceEnabled()) {
-                  log.trace("Creating proxy bean for injection point: " + injectionPoint.toString());
-               }
-               addInjectableClientProxyBean((Field) member, (MicroserviceReference) qualifier, preProcessQualifiers(injectionPoint.getQualifiers()), annotations, beanManager);
-               this.injectionPointsCount = this.injectionPointsCount + 1;
-            }
-         });
-      }
-   }
-
-   private Set<Annotation> preProcessQualifiers(final Set<Annotation> qualifiers) {
-      final Set<Annotation> processed = new HashSet<>(qualifiers);
-      final List<String> current = qualifiers.stream().map(q -> q.annotationType().getName()).collect(Collectors.toList());
+      bean.getInjectionPoints().stream().filter(MicroservicesCDIExtension::hasMicroserviceReference).forEach(injectionPoint -> {
+         if (log.isTraceEnabled()) {
+            log.trace("Creating proxy bean for injection point: " + injectionPoint.toString());
+         }
 
+         MicroserviceProxy microserviceProxy = new MicroserviceProxyAnnotationLiteral(injectionPoint);
+         injectionPoint.getQualifiers().add(microserviceProxy);
 
-      if (!current.contains(Any.class.getName())) {
-         processed.add(new AnyAnnotationLiteral());
-         current.add(Any.class.getName());
```

Comment:
It has been moved to `MicroserviceProxyBean`, see its [prepareQualifiers](https://github.com/livthomas/SilverWare/blob/1e863807df8a84148cde2959d90bb4c904ce285d/cdi-microservice-provider/src/main/java/io/silverware/microservices/providers/cdi/internal/MicroserviceProxyBean.java#L112-L127) method. The main reason for this was not to pass too many arguments to the constructor of this class. At the same time I think it is better to make modifications needed just for this class inside it.